src/main/java/com/mraof/minestuck/entity/item/ReturningProjectileEntity.java

```diff
@@ -0,0 +1,105 @@
+package com.mraof.minestuck.entity.item;
+
+import com.mraof.minestuck.client.renderer.entity.RendersAsItem;
+import com.mraof.minestuck.item.MSItems;
+import net.minecraft.entity.*;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.entity.player.ServerPlayerEntity;
+import net.minecraft.entity.projectile.ProjectileItemEntity;
+import net.minecraft.item.Item;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.network.IPacket;
+import net.minecraft.util.DamageSource;
+import net.minecraft.util.SoundCategory;
+import net.minecraft.util.SoundEvents;
+import net.minecraft.util.math.*;
+import net.minecraft.world.World;
+import net.minecraftforge.fml.network.NetworkHooks;
+
+public class ReturningProjectileEntity extends ProjectileItemEntity implements RendersAsItem
+{
+	private int cooldownTicks;
+	private int bounce;
```

Comment:
Are these uninteresting to save? Or is this entity not meant to be serialized?

Reply:
Serialized as in saved to/loaded from the world file

Reply:
I do not understand the importance well enough to know

Reply:
Serialized!